bioptim/examples/getting_started/example_variable_scaling.py

```diff
@@ -0,0 +1,152 @@
+
+One scaling should be declared for each phase for the states and controls. The scaling of the parameters should be
+declared in the parameter declaration like in the example getting_started/custom_parameters.py.
+"""
+
+import biorbd_casadi as biorbd
+from bioptim import (
+    OptimalControlProgram,
+    DynamicsFcn,
+    Dynamics,
+    Bounds,
+    QAndQDotBounds,
+    InitialGuess,
+    ObjectiveFcn,
+    Objective,
+    OdeSolver,
+    CostType,
+    Solver,
+    VariableScalingList,
+)
+
+
+def prepare_ocp(
+    biorbd_model_path: str,
+    final_time: float,
+    n_shooting: int,
+    ode_solver: OdeSolver = OdeSolver.RK4(),
+    use_sx: bool = True,
+    n_threads: int = 1,
+) -> OptimalControlProgram:
+    """
+    The initialization of an ocp
+
+    Parameters
+    ----------
+    biorbd_model_path: str
+        The path to the biorbd model
+    final_time: float
+        The time in second required to perform the task
+    n_shooting: int
+        The number of shooting points to define int the direct multiple shooting program
+    ode_solver: OdeSolver = OdeSolver.RK4()
+        Which type of OdeSolver to use
+    use_sx: bool
+        If the SX variable should be used instead of MX (can be extensive on RAM)
+    n_threads: int
+        The number of threads to use in the paralleling (1 = no parallel computing)
+
+    Returns
+    -------
+    The OptimalControlProgram ready to be solved
+    """
+
+    biorbd_model = biorbd.Model(biorbd_model_path)
+
+    # Add objective functions
+    objective_functions = Objective(ObjectiveFcn.Lagrange.MINIMIZE_CONTROL, key="tau")
+
+    # Dynamics
+    dynamics = Dynamics(DynamicsFcn.TORQUE_DRIVEN)
+
+    # Path constraint
+    x_bounds = QAndQDotBounds(biorbd_model)
+    x_bounds[:, [0, -1]] = 0
+    x_bounds[1, -1] = 3.14
+
+    # Initial guess
+    n_q = biorbd_model.nbQ()
+    n_qdot = biorbd_model.nbQdot()
+    x_init = InitialGuess([0] * (n_q + n_qdot))
+
+    # Define control path constraint
+    n_tau = biorbd_model.nbGeneralizedTorque()
+    tau_min, tau_max, tau_init = -100, 100, 0
+    u_bounds = Bounds([tau_min] * n_tau, [tau_max] * n_tau)
+    u_bounds[1, :] = 0
+
+    u_init = InitialGuess([tau_init] * n_tau)
+
+    # Variable scaling
+    # x_scaling = VariableScalingList()
+    # x_scaling.add("q", scaling=[1, 3])  # declare keys in order, so that they are concatenated in the right order
+    # x_scaling.add("qdot", scaling=[100, 30])
+    #
+    # xdot_scaling = VariableScalingList()
+    # xdot_scaling.add("qdot", scaling=[100, 30])
+    # xdot_scaling.add("qddot", scaling=[100, 30])
+    #
+    # u_scaling = VariableScalingList()
+    # u_scaling.add("tau", scaling=[1, 1])
+
+
+    x_scaling = VariableScalingList()
+    x_scaling.add("q", scaling=[1, 1])  # declare keys in order, so that they are concatenated in the right order
+    x_scaling.add("qdot", scaling=[1, 1])
+
+    xdot_scaling = VariableScalingList()
```

Comment:
why scaling xdot?
Is is for "implicit" dynamics with additional algebraic states ?